apps/rush-lib/src/logic/ApprovedPackagesChecker.ts

```diff
@@ -17,30 +38,41 @@ export class ApprovedPackagesChecker {
    *
    * If the "approvedPackagesPolicy" feature is not enabled, then no action is taken.
    */
-  public static rewriteConfigFiles(rushConfiguration: RushConfiguration): void {
-    const approvedPackagesPolicy: ApprovedPackagesPolicy = rushConfiguration.approvedPackagesPolicy;
-    if (!approvedPackagesPolicy.enabled) {
-      return;
+  public rewriteConfigFiles(): void {
+    const approvedPackagesPolicy: ApprovedPackagesPolicy = this._rushConfiguration.approvedPackagesPolicy;
+    if (approvedPackagesPolicy.enabled) {
+      approvedPackagesPolicy.browserApprovedPackages.saveToFile();
+      approvedPackagesPolicy.nonbrowserApprovedPackages.saveToFile();
     }
+  }
 
-    for (const rushProject of rushConfiguration.projects) {
+  private _updateApprovedPackagesPolicy(): void {
+    for (const rushProject of this._rushConfiguration.projects) {
       const packageJson: IPackageJson = rushProject.packageJson;
 
-      ApprovedPackagesChecker._collectDependencies(packageJson.dependencies,
-        approvedPackagesPolicy, rushProject);
-      ApprovedPackagesChecker._collectDependencies(packageJson.optionalDependencies,
-        approvedPackagesPolicy, rushProject);
-      ApprovedPackagesChecker._collectDependencies(packageJson.devDependencies,
-        approvedPackagesPolicy, rushProject);
+      this._collectDependencies(
+        packageJson.dependencies,
+        this._approvedPackagesPolicy,
+        rushProject
+      );
+      this._collectDependencies(
+        packageJson.optionalDependencies,
+        this._approvedPackagesPolicy,
+        rushProject
+      );
+      this._collectDependencies(
```

Comment:
@iclanton  Seems like we should have included `peerDependencies` here as well.  That seems like a relatively low risk change, so I'd be okay with including it in your PR (but with a separate change log entry).